### PR TITLE
[BH-1770][BH-1677] Update fsl drivers

### PR DESCRIPTION
**Pull request by dawidwojtas-mudita**
Now the system uses 2.14.0 version.
Add only necessary drivers:
- pit
- pmu
- rtwdog
- snvs_hp
- snvs_lp
- src
- trng
- wdog
- gpio
- gpt
- lpi2c
- lpuart
- lpuart_edma
- common
- common_arm
- dmamux
- edma
- flexram
- flexram_allocate
- clock
- cache
- CMSIS
- dcp
- iomuxc

### CODE REVIEW

File: hal/src/emmc.c [hunk 1]
```diff
@@ -1,50 +1,225 @@
...
+//#define DEBUG_USDHC
+#ifdef DEBUG_USDHC
+#include <drivers/fsl_src.h>
+
+#include "log.h"
+#include "eink.h"
+
+static void reportUSDHCRegisterSerial(void);
+static void reportUSDHCRegistersEink(status_t error);
+#endif
```
Comment (Maciej-Mudita) [inline hal/src/emmc.c:19]: I wonder if it is necessary?

File: platform/CMakeLists.txt [hunk 2]
```diff
@@ -38,6 +38,7 @@ target_compile_definitions(${LIB_NAME} PUBLIC
     CPU_MIMXRT1051DVL6B_cm7
     CPU_MIMXRT1051DVL6B
     __FPU_PRESENT
+    __MCUXPRESSO
```
Comment (Maciej-Mudita) [inline platform/CMakeLists.txt:41]: do you know why this is needed?

**Review by Maciej-Mudita — APPROVED** [review]

File: platform/CMakeLists.txt (hunk 2 above)
Comment (dawidwojtas-mudita) [reply platform/CMakeLists.txt:41]: The `system_MIMXRT1051.c` uses this define for the init vector table in `SystemInit()` function.
```
#if defined(__MCUXPRESSO)
    extern uint32_t g_pfnVectors[];  // Vector table defined in startup code
    SCB->VTOR = (uint32_t)g_pfnVectors;
#endif
```

File: hal/src/emmc.c (hunk 1 above)
Comment (dawidwojtas-mudita) [reply hal/src/emmc.c:19]: I used this code for debugging issues with eMMC and USDHC drivers. I decided to leave this code for the future. Who knows maybe some new issues will be discovered ;)

**Review by Lefucjusz — APPROVED** [review]

**Pull request merged**